src/lib.rs

```diff
@@ -470,6 +481,16 @@ mod tests {
             assert!((*settings_ref).rho == 0.7, "Expected rho to be 0.7 but found {}", (*settings_ref).rho);
 
         }
+
+        // Test `update_settings()`:
```

Comment:
Can we introduce this as a new test instead of modifying an old one? Perhaps we call it `update_prob_settings` with the comment `Update some settings using update_settings on the Problem object`, and rename the old `update_prob_settings` as `update_rho` (with the comment `Update a setting using a method generated by the settings macro`.